packages/core/src/interpreter.ts

```diff
@@ -534,6 +535,7 @@ export class Interpreter<
     this.scheduler.clear();
     this.initialized = false;
     this.status = InterpreterStatus.Stopped;
+    this._initialState = undefined;
```

Comment:
Atm the `service.initialState` is being cached:
https://github.com/statelyai/xstate/blob/4bb3a91f232599034a25159e89f5c82ce9b3776d/packages/core/src/interpreter.ts#L213-L222
and it should stay cached, otherwise the `useSyncExternalStore` complains about `getSnapshot` not yielding stable results when `service.status === InterpreterStatus.NotStarted`.

However, this has to be reset in the same scenario as mentioned here:
https://github.com/statelyai/xstate/pull/2939/files#r784159286